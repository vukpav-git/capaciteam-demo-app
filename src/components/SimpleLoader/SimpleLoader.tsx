import { CircularProgress } from "@mui/material";

interface SimpleLoaderProps {
  color?: string;
  marginTop?: string;
  marginBottom?: string;
  marginLeft?: string;
  marginRight?: string;
}

const SimpleLoader = ({
  marginTop,
  marginBottom,
  marginLeft,
  marginRight,
}: SimpleLoaderProps) => (
  <CircularProgress
    data-testid="circular-progress"
    size={40}
    sx={{
      color: "black",
      marginTop,
      marginLeft,
      marginBottom,
      marginRight,
    }}
  />
);

export default SimpleLoader;
