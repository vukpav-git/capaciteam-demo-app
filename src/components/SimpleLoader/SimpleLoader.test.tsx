import { render, screen } from "@testing-library/react";

import SimpleLoader from "./SimpleLoader";

describe("SimpleLoader tests", () => {
  const testProps = {
    color: "black",
    marginTop: "20px",
    marginBottom: "20px",
    marginLeft: "20px",
    marginRight: "20px",
  };

  it("should match snapshot", () => {
    const { asFragment } = render(<SimpleLoader {...testProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the CircularProgress with correct styles", () => {
    render(<SimpleLoader {...testProps} />);
    const loader = screen.getByTestId("circular-progress");

    expect(loader).toBeInTheDocument();
    expect(loader).toHaveStyle(`color: ${testProps.color}`);
    expect(loader).toHaveStyle(`marginTop: ${testProps.marginTop}`);
    expect(loader).toHaveStyle(`marginBottom: ${testProps.marginBottom}`);
    expect(loader).toHaveStyle(`marginLeft: ${testProps.marginLeft}`);
    expect(loader).toHaveStyle(`marginRight: ${testProps.marginRight}`);
  });
});
