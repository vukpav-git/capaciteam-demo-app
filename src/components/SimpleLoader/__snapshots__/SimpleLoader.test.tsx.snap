// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleLoader tests should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-u8tkwc-MuiCircularProgress-root"
    data-testid="circular-progress"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</DocumentFragment>
`;
